From: Andy Gospodarek <gospo@broadcom.com>
Date: Fri, 3 Nov 2017 03:32:38 -0400
Subject: bnxt_en: fix typo in bnxt_set_coalesce
Patch-mainline: v4.15-rc1
Git-commit: de4a10ef6eff0eb0ced97a39dc3edd0d3101b6ed
References: bsc#1050242 FATE#322914

Recent refactoring of coalesce settings contained a typo that prevents
receive settings from being set properly.

Fixes: 18775aa8a91f ("bnxt_en: Reorganize the coalescing parameters.")
Signed-off-by: Andy Gospodarek <gospo@broadcom.com>
Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt_ethtool.c
@@ -84,7 +84,7 @@ static int bnxt_set_coalesce(struct net_
 	hw_coal->coal_ticks_irq = coal->rx_coalesce_usecs_irq;
 	hw_coal->coal_bufs_irq = coal->rx_max_coalesced_frames_irq * mult;
 
-	hw_coal = &bp->rx_coal;
+	hw_coal = &bp->tx_coal;
 	mult = hw_coal->bufs_per_record;
 	hw_coal->coal_ticks = coal->tx_coalesce_usecs;
 	hw_coal->coal_bufs = coal->tx_max_coalesced_frames * mult;
